catalog/app/constants/routes.js

```diff
@@ -114,9 +120,9 @@ export const bucketPackageDetail = {
 export const bucketPackageTree = {
   path: `/b/:bucket/packages/:name(${PACKAGE_PATTERN})/tree/:revision/:path(.*)?`,
   url: (bucket, name, revision, path = '') =>
-    revision === 'latest' && !path
+    (!revision || revision === 'latest') && !path
```

Comment:
I've lost. Do we use both `revision`s and `hash`es, or they are the same?

Reply:
Positive (non-negative :) ) conditions are easier to read:

```
(path && revision !== 'latest')
 ? `/b/${bucket}/packages/${name}/tree/${revision || 'latest'}/${encode(path)}`
 : bucketPackageDetail.url(bucket, name)
```

Reply:
> Do we use both revisions and hashes, or they are the same?

well, it may be a little confusing, but i think conceptually "revision" is a "package instance" identified by a hash, and in this particular case it makes sense to keep the argument name generic enough ("revision"), bc it can be either a hash or some other thing that identifies a revision, such as a tag (e.g. `latest`) or a timestamp aka pointer file